primap2/csg/_wrapper.py

```diff
@@ -0,0 +1,132 @@
+        it early (i.e. with high priority) before the more general rules which should
+        be applied for all other countries.
+        You can also specify the "entity" or "variable" in the selection, which will
+        limit the rule to a specific entity or variable, respectively. For each
+        DataArray in the input_data Dataset, the variable is its name, the entity is
+        the value of the key `entity` in its attrs.
+    strategy_definition
+        Defines the filling strategies to be used when filling timeseries with other
+        timeseries. Again, the priority is defined by a list of selections and
+        corresponding strategies which are used "from left to right". Selections can use
+        any dimension and don't have to apply to only one timeseries. For example, to
+        define a default strategy which should be used for all timeseries unless
+        something else is configured, configure an empty selection as the last
+        (rightmost) entry.
+        You can also specify the "entity" or "variable" in the selection, which will
+        limit the rule to a specific entity or variable, respectively. For each
+        DataArray in the input_data Dataset, the variable is its name, the entity is
+        the value of the key `entity` in its attrs.
+    result_prio_coords
+        Defines the vales for the priority coordinates in the output dataset. As the
+        priority coordinates differ for all input sources there is no canonical vale
+        for the result and it has to be explicitly defined
+    limit_coords
+        Optional parameter to remove data for coordinate vales not needed for the
+        composition from the input data. The time coordinate is treated separately.
+    time_range
+        Optional parameter to limit the time coverage of the input data. Currently
+        only (year_from, year_to) is supported
+    metadata
+        Set metadata values such as title and references
+    progress_bar
+        By default, show progress bars using the tqdm package during the
+        operation. If None, don't show any progress bars. You can supply a class
+        compatible to tqdm.tqdm's protocol if you want to customize the progress bar.
+
+    Returns
+    -------
+        xr.Dataset with composed data according to the given priority and strategy
+        definitions
+
+    """
+
+    # limit input data to these values
+    if limit_coords is not None:
+        if "variable" in limit_coords.keys():
+            variables = limit_coords["variable"]
+            limit_coords.pop("variable")
+            input_ds = input_ds[variables].pr.loc[limit_coords]
+
+        else:
+            input_ds = input_ds.pr.loc[limit_coords]
+
+    # set time range according to input
+    if time_range is not None:
+        input_ds = input_ds.pr.loc[
+            {"time": pd.date_range(time_range[0], time_range[1], freq="YS", inclusive="both")}
```

Comment:
I thought we can make it more versatile later. But I think I can do that now because I've spent the last two weeks fighting with time ranges in xarray, so It should be a quick fix.

Reply:
I would limit the option to slice and datetime index. When using the datetime index users would have to make sure the values actually exist.
For the composite source generator the only use case I can think of is a range though. But it's good if it's more versatile than just start and end year and e.g. the interval can be specified.